tests/drivers/counter/counter_basic_api/src/test_counter.c

```diff
@@ -730,8 +731,15 @@ static void test_valid_function_without_alarm(const struct device *dev)
 		ticks_expected = counter_us_to_ticks(dev, wait_for_us);
 	}
 
+#ifdef COUNTER_HAS_INACCURACY
+	/* Some counters such as the NXP LPC RTC have greater inaccuracy.
+	 * Set 30% or 2 ticks tolerance, whichever is greater
+	 */
+	ticks_tol = (ticks_expected / 10) * 3;
+#else
 	/* Set 10% or 2 ticks tolerance, whichever is greater */
 	ticks_tol = ticks_expected / 10;
```

Comment:
Here it would be something like:
`ticks_tol = MAX(2, (ticks_expected * COUNTER_ACCEPTED_ACCURACY) / 100);`